src/internal/ajax/AjaxResponse.ts

```diff
@@ -76,6 +87,27 @@ export class AjaxResponse<T> {
     const { status, responseType } = xhr;
     this.status = status ?? 0;
     this.responseType = responseType ?? '';
+
+    // Parse the response headers in advance for the user. There's really
+    // not a great way to get all of them. So we need to parse the header string
+    // we get back. It comes in a simple enough format:
+    //
+    // header-name: value here
+    // content-type: application/json
+    // other-header-here: some, other, values, or, whatever
+    const allHeaders = xhr.getAllResponseHeaders();
+    this.responseHeaders = allHeaders
+      ? // Split the header text into lines
+        allHeaders.split('\n').reduce((headers: Record<string, string>, line) => {
+          // Split the lines on the first ": " as
+          // "key: value". Note that the value could
+          // technically have a ": " in it.
+          const index = line.indexOf(': ');
+          headers[line.slice(0, index)] = line.slice(index + 2);
+          return headers;
+        }, {})
+      : {};
```

Comment:
Maybe we could implement `responseHeaders` as a getter so that the headers don't have to be parsed for every response?

Reply:
I did that initially, but then I had a add a private property, tests got messy, I got lazy because it was 2am, etc. Happy to take another pass at that in a follow up PR, though.